tablo/interfaces/arcgis/views.py

```diff
@@ -206,37 +215,45 @@ class QueryView(FeatureLayerView):
     def handle_request(self, request, **kwargs):
 
         search_params = {}
-        return_ids_only = kwargs.get('returnIdsOnly', 'false') in ('true', 'True')
+        return_ids_only = kwargs.get('returnIdsOnly', 'false').lower() == 'true'
 
         if 'where' in kwargs and kwargs['where'] != '':
             search_params['additional_where_clause'] = kwargs.get('where')
 
         if 'time' in kwargs and kwargs['time'] != '':
             start_time, end_time = kwargs['time'].split(',')
-            search_params['start_time'] = time.strftime('%Y-%m-%d %H:%M:%S', time.gmtime(float(start_time)/1000))
-            search_params['end_time'] = time.strftime('%Y-%m-%d %H:%M:%S', time.gmtime(float(end_time)/1000))
+            search_params['start_time'] = time.strftime('%Y-%m-%d %H:%M:%S', time.gmtime(float(start_time) / 1000))
+            search_params['end_time'] = time.strftime('%Y-%m-%d %H:%M:%S', time.gmtime(float(end_time) / 1000))
 
         search_params['out_sr'] = kwargs.get('outSR')
 
         if 'objectIds' in kwargs:
-            search_params['object_ids'] = kwargs.get('objectIds', '').split(',')
+            search_params['object_ids'] = (kwargs['objectIds'] or '').split(',')
 
         if kwargs.get('geometryType') == 'esriGeometryEnvelope':
             search_params['extent'] = Extent(json.loads(kwargs['geometry']))
 
         if kwargs.get('outFields') and not return_ids_only:
-            search_params['return_fields'] = kwargs.get('outFields').split(',')
+            search_params['return_fields'] = (kwargs['outFields'] or '').split(',')
+
+        if kwargs.get('orderByFields') and not return_ids_only:
+            search_params['order_by_fields'] = (kwargs['orderByFields'] or '').split(',')
 
         if return_ids_only:
             search_params['return_fields'] = [self.feature_service_layer.object_id_field]
-        elif kwargs.get('returnCountOnly', 'false') == 'true':
-            search_params['only_return_count'] = True
-        elif kwargs.get('returnGeometry', 'true') == 'true':
-            search_params['return_geometry'] = True
+            search_params['return_geometry'] = False
+        elif kwargs.get('returnCountOnly', 'false').lower() == 'true':
+            search_params['count_only'] = True
+        elif kwargs.get('returnGeometry', 'true').lower() == 'false':
+            search_params['return_geometry'] = False
 
         try:
-            query_response = self.feature_service_layer.perform_query(**search_params)
-        except ValueError:
+            query_response = self.feature_service_layer.perform_query(
+                limit=kwargs.get('limit', 1000),
```

Comment:
I did two things:

1. Flattened the use of limit/offset in `models`: the code now applies them in `perform_query`, and the model-level defaults for them are defined in the `perform_query` method signature.
2. Added `query_limit_default` as a class variable in `QueryView`: this means others may extend and override the class for difference of behavior rather than having to copy/adapt the whole method (like we often have to do in extended TastyPie resources).

These two changes better reflect the fact that `QueryView` and `FeatureServiceLayer.perform_query` may have different performance concerns when it comes to  limit/offset. It also keeps class- and method- specific considerations bound to the classes and methods themselves.